### DISCUSSION TITLE
A `file-in-worktree` feature - jump to the same file in different worktrees.

### DISCUSSION THREAD

**Discussion by cstml** [magit/magit#4959] [Ideas]
It would be a great feature if:
- given a file open in the current buffer,
one can (via `file-in-worktree`) see open another buffer with the same file, at the same position (line number) but in a different worktree. 

WDYT? 

I'm willing to give it a go to implement it, wondering if other people would find it useful. 

The inspiration is:
- [Git's Best And Most Unknown Feature - The Primagean's video](https://youtu.be/2uEqYw-N8uE?t=320) highlighting a similar feature.

**Comment by tarsius** [comment]
Are you aware that you can already "visit another version" of the current file using `magit-find-file`? That command first prompts for a revision and then a file. If you are visiting a file, then that is the default at the second prompt and you can just confirm it. This takes care of jumping to the same line if possible. It does not just jump to the same line number, but also takes removed and added lines before the current line into account.

Granted this is not exactly what you are asking for. This visits a *blob*, which isn't visiting a file that is checked out somewhere. Never-the-less this is sufficient in many cases, maybe it is for your use-cases as well.

Checking out the appropriate commit in a worktree certainly would be useful too, but there isn't just one way of doing that. Should this only offer opening "the same file" in an *existing worktree*, or should it also offer to create an ad hoc worktree? Where should that worktree be created? Should a new branch be created or should the commit be checked out directly, creating a detached HEAD? What if there are multiple branches that point at the commit? Depending on their use-cases, different users will answer these questions differently.

So it does get complicated and while I likely would not implement all possible paths in a first version, I would want to think about these options more and keep the door open for implementing them later without requiring too much refactoring before taking further steps.

It might take me a while until I get around to that, especially because I have been planning to make big changes to the whole "same position in different files/blogs/commits/trees/buffers/diffs/logs..." anyway.

At the same time I can of course see that a minimal viable version can already be very useful too, so I encourage you to implement one that works for you. But I cannot guarantee that I would merge it.

**Reply by cstml** [reply]
Thank you for taking the time to respond. The `magit-find-file` is more than good enough, if not precisely the tool I was looking for. Otherwise, fuzzy finding the file in the worktree is probably a more adequate solution, and out of scope in this case.
